src/components/alarm/AlarmController.cpp

```diff
@@ -0,0 +1,114 @@
+/*  Copyright (C) 2021 mruss77, Florian
+
+    This file is part of InfiniTime.
+
+    InfiniTime is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    InfiniTime is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+#include "AlarmController.h"
+#include "systemtask/SystemTask.h"
+#include "app_timer.h"
+#include "task.h"
+#include <chrono>
+
+using namespace Pinetime::Controllers;
+using namespace std::chrono_literals;
+
+AlarmController::AlarmController(Controllers::DateTime& dateTimeController) : dateTimeController {dateTimeController} {
+}
+
+APP_TIMER_DEF(alarmAppTimer);
+
+namespace {
+  void SetOffAlarm(void* p_context) {
+    auto* controller = static_cast<Pinetime::Controllers::AlarmController*>(p_context);
+    if (controller != nullptr) {
+      controller->SetOffAlarmNow();
+    }
+  }
+}
+
+void AlarmController::Init(System::SystemTask* systemTask) {
+  app_timer_create(&alarmAppTimer, APP_TIMER_MODE_SINGLE_SHOT, SetOffAlarm);
+  this->systemTask = systemTask;
+}
+
+void AlarmController::SetAlarm(uint8_t alarmHr, uint8_t alarmMin) {
+  hours = alarmHr;
+  minutes = alarmMin;
+  state = AlarmState::Set;
+  ScheduleAlarm();
+}
+
```

Comment:
If I change the time without enabling the alarm, the new time won't be saved. Perhaps this function could just set the time, and ScheduleAlarm() could be made public.

Reply:
Agreed - changed SetAlarm to SetAlarmTime and made ScheduleAlarm public.